Open WorldCat.js

```diff
@@ -341,7 +340,17 @@ function fetchIDs(isbns, ids, callback) {
 				} else {
 					Z.debug("Could not extract OCLC ID for ISBN " + isbn);
 				}
-			} else {
+			}
+			//but sometimes we have single items
+			else if (getFirstContextObj(doc)) {
```

Comment:
Isn't it here enough to check for the presence of this canonical url, i.e. something like
```js
} else {
   var canonicalURL = ZU.xpathText(doc, '/html/head/link[@rel="canonical"]/@href');
   if (canonicalURL) {
      oclcID = extractOCLCID(canonicalURL);
      if(!oclcID) throw new Error("WorldCat: Failed to extract OCLC ID from URL: " + url);
         scrape([oclcID]);
      }
   } else {
      Z.debug("No search results found for ISBN " + isbn);
   }
}
```
?